src/corehost/common/pal.windows.cpp

```diff
@@ -347,14 +385,68 @@ bool pal::clr_palstring(const char* cstr, pal::string_t* out)
 
 bool pal::realpath(string_t* path)
 {
+
+    if (!LongFile::ShouldNormalize(*path))
+    {
+        return true;
+    }
+
     char_t buf[MAX_PATH];
-    auto res = ::GetFullPathNameW(path->c_str(), MAX_PATH, buf, nullptr);
-    if (res == 0 || res > MAX_PATH)
+    auto size = ::GetFullPathNameW(path->c_str(), MAX_PATH, buf, nullptr);
+    
+    if (size == 0)
     {
         trace::error(_X("Error resolving full path [%s]"), path->c_str());
         return false;
     }
-    path->assign(buf);
+
+    if (size < MAX_PATH)
+    {
+        path->assign(buf);
+        return true;
+    }
+
+    const string_t* prefix = &LongFile::ExtendedPrefix;
```

Comment:
This function looks overcomplicated and hard to read. I think that it would be better to use string_t instead of vector, just resize str to the size returned by the first call to `GetFullPathNameW`, then call the `GetFullPathNameW` with `str.data()` and finally prepend the prefix if needed using string_t::insert. Without all the prefix len computations.